codec/reflectcodec/type_codec.go

```diff
@@ -318,12 +349,25 @@ func (c *genericCodec) marshal(value reflect.Value, p *wrappers.Packer, maxSlice
 		p.PackBool(value.Bool())
 		return p.Err
 	case reflect.Ptr:
-		if value.IsNil() { // Can't marshal nil (except nil slices)
+		isNil := value.IsNil()
+		if omitEmpty {
+			p.PackBool(isNil)
+			if isNil {
+				return p.Err
+			}
+		} else if isNil {
 			return errMarshalNil
 		}
+
 		return c.marshal(value.Elem(), p, c.maxSliceLen)
 	case reflect.Interface:
-		if value.IsNil() { // Can't marshal nil (except nil slices)
+		isNil := value.IsNil()
+		if omitEmpty {
+			p.PackBool(isNil)
+			if isNil {
+				return p.Err
```

Comment:
shouldn't we also check `p.Err` in the case where `isNil == false`?

Reply:
Fixed in e430e5efd